en_US/faq/use-guide.md

```diff
@@ -417,3 +417,141 @@ Client disconnect link error code list:
+
+### Keyword: “reason: {shutdown, kicked}”
+
+Indicates that the client connection was manually kicked out (click the Kick Out button on the Dashboard or call the `DELETE clients/{clientid}` API). If the original client is still connected, EMQX will send a DISCONNECT packet with Reason Code 0x98 (Administrative action), and then close the old connection.
+
+### Keyword: “reason: {shutdown, tcp_closed}”
+
+This log indicates that the client directly closed the network connection without sending a DISCONNECT packet. The corresponding reason would be `{shutdown, normal}`, which means that the client first sent a DISCONNECT packet with a Reason Code of 0 and then closed the network connection.
+
+Therefore, when you encounter that `tcp_closed` appears in the EMQX logs, and this disconnection behavior is not what you expected, it is recommended to investigate the client's implementation for any potential issues.
+
+### Keyword: “maximum heap size reached”
+
+The appearance of this log indicates that the heap occupied by the current client process has exceeded the preset maximum value, and the client process will be forcibly killed to ensure the availability of EMQX, and avoid the unlimited growth of the memory usage of the client process and eventually lead to EMQX OOM. 
+
+Usually, message accumulation is the main reason for the increase in client process heap occupation, and message accumulation is generally because the client's consumption capacity does not match the peer's message production capacity. The recommended solution is to optimize the client processing code or use shared subscriptions to disperse the load.
+
+The related configuration item is `force_shutdown_policy`, and its format is `<Maximum Message Queue Length>|<Maximum Heap Size>`, for example, `10000|64MB`.  `<Maximum Heap Size>` refers to the maximum heap memory that each client process can occupy.
+
+### Keyword: “Parse failed for function_clause”
+
+This log indicates that the parsing of the packet failed. Maybe because it is not an MQTT packet, we have encountered many situations where HTTP requests are sent to the MQTT port, or because the packet contains non-UTF-8 characters. We can search the `Frame data` keyword in this "Parse failed..." log to view the complete message and help us analyze the possible reasons for the parsing failure.
+
+### Keyword: “Dropped msg due to mqueue is full”
+
+EMQX can send multiple unconfirmed QoS 1 and QoS 2 messages at the same time, but limited by the performance of the client, we usually limit the maximum number of messages allowed to be sent at the same time. After reaching this limit, subsequent arriving messages will be cached by EMQX in the message queue of each client process.
```

Comment:
```suggestion
EMQX can send multiple unconfirmed QoS 1 and QoS 2 messages simultaneously. However, to prevent excessive message buffering, there is a maximum length limit for the message queue. When reaching this limit, subsequent arriving messages will be cached by EMQX in the message queue of each client process.
```

Reply:
The original meaning is changed here. I made the following changes:

```
EMQX can send multiple unconfirmed QoS 1 and QoS 2 messages simultaneously, but limited by the performance of the client, we usually limit the maximum number of messages in flight. After reaching this limit, subsequent arriving messages will be cached by EMQX in the message queue of each client process.

However, the message queus still has a length limit. So If the message queue is full when a new message arrives, the latest message will be enqueued, but the oldest message in the queue will be dequeued and dropped. EMQX generates the log message: "Dropped msg due to mqueue is full."
```

Reply:
OK, just made a little editing:

"EMQX can send multiple unconfirmed QoS 1 and QoS 2 messages simultaneously, but limited by the performance of the client, we usually limit the maximum number of messages in flight. After reaching this limit, subsequent incoming messages will be cached by EMQX in the message queue of each client process.

However, the message queues still have a length limit. If the message queue is full when a new message arrives, the most recent message will be enqueued, but the oldest message in the queue will be dequeued and discarded. EMQX generates the log message: "Dropped msg due to mqueue is full."

Reply:
Updated, I still use 'dropped' rather than 'discarded', keep it consistent with the log.